### PR TITLE
feat: uikit snackbar and MessengerService

### PR DESCRIPTION
**Pull request by nesquikm**
## Description

copilot:all

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/app/service/messenger/cubit/messenger_cubit.dart [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import 'package:app/app/service/messenger/service/service.dart';
+import 'package:app/di/di.dart';
+import 'package:bloc/bloc.dart';
+import 'package:clock/clock.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'messenger_state.dart';
+part 'messenger_cubit.freezed.dart';
+
+class MessengerCubit extends Cubit<MessengerState> {
```
Comment (Alex-A4) [inline lib/app/service/messenger/cubit/messenger_cubit.dart:11]: Docs pls

File: lib/app/service/messenger/service/messenger_service.dart [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import 'dart:async';
+
+import 'package:app/app/service/messenger/message.dart';
+import 'package:injectable/injectable.dart';
+
+@singleton
+class MessengerService {
```
Comment (Alex-A4) [inline lib/app/service/messenger/service/messenger_service.dart:7]: Docs pls

File: lib/app/service/messenger/widget/messenger_service_widget.dart [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  void _showNextMessage() {
+    // Wait for the current snackbar to be dismissed
+    if (_isSnackbarShown) {
+      return;
+    }
+
+    // We can't use current context here because it's not contains Navigator
+    final ctx = NavigationService.navigatorKey.currentState?.context;
```
Comment (Alex-A4) [inline lib/app/service/messenger/widget/messenger_service_widget.dart:42]: Why like static method? It's in `GetIt`

File: lib/app/service/messenger/cubit/messenger_cubit.dart [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    if (previousMessageTime != null &&
+        now.difference(previousMessageTime) < message.debounceTime) {
+      return;
+    }
+
+    // Put message to the start of the queue
+    emit(
+      state.copyWith(
+        messagesToShow: [
+          message,
```
[the remark this answers is not in the document]
Comment (nesquikm) [reply lib/app/service/messenger/cubit/messenger_cubit.dart:41]: Yep, thnx!

File: lib/app/service/messenger/service/messenger_service.dart (hunk 2 above)
Comment (nesquikm) [reply lib/app/service/messenger/service/messenger_service.dart:10]: Done!

File: lib/app/service/messenger/widget/messenger_service_widget.dart (hunk 3 above)
Comment (nesquikm) [reply lib/app/service/messenger/widget/messenger_service_widget.dart:48]: Let's just leave it as it is. Temporary :)

**Review by Alex-A4 — APPROVED** [review]

**Pull request merged**